### ISSUE TITLE
Game freezes in singleplayer

### ISSUE THREAD

**Issue by Tinter** [IDI-Systems/acre2#1157]
**Mods (complete and add to the following information):**
- **Arma 3:** `2.06.148221` 
- **CBA:** `3.15.6` 
- **ACRE2:** `2.9.1.1045` 

**Description:**
When playing the game in single player, the game will freeze.

**Steps to reproduce:**
- Open Altis in the editor
- Click preview
- Game will freeze within 5 minutes, mostly before 2.

**Expected behavior:**
No freeze.

**Where did the issue occur?**
-  Singleplayer / Editor (Singleplayer) / Editor (Multiplayer) 
- But not in multiplayer on a dedicated server

**Log Files:**
- RPT file: https://gist.github.com/Tinter/3439431494430b12e282d0f872c52fc7
- ACRE log: https://gist.github.com/Tinter/eb065d23dc199b4f488b2d1dfb75f56b

**Additional context:**
I am running the game on Arch Linux, through Proton GE normally. I can't see specific versions because my steam library won't load, but it appears through several Proton versions both several GE versions and official Proton versions.
Recreated with just CBA and ACRE.
No freeze appears with just CBA.
The freeze is so bad that sigterm will not close the game, it needs to be sigkill.
I first saw the issue around the 8th of October 2021, but did not isolate it to ACRE until today. Game would work fine without  freezing with mods for months before October.

**Comment by Tinter** [comment]
It was suggested this is less in singleplayer, and more when I am the server, which makes sense as it appears in the multiplayer editor.

**Comment by jonpas** [comment]
Looking at the logs, ACRE seems to initialize just fine, even loads the map. Must be something with libraries (Proton etc.) that changed and broke it.

You said you first saw the issue around 8th October, that was before ACRE2 v2.9.1 update, and there were no changes for a good while before that.

Also mind, we don't officially support Proton, but we do as much as we can in not breaking it and accepting any pull requests that may improve the state.

**Comment by Tinter** [comment]
> Looking at the logs, ACRE seems to initialize just fine, even loads the map. Must be something with libraries (Proton etc.) that changed and broke it.
> 
> You said you first saw the issue around 8th October, that was before ACRE2 v2.9.1 update, and there were no changes for a good while before that.

It looks like it coincided just about with the latest Arma update on October 5th.
https://dev.arma3.com/post/spotrep-00100 

> Also mind, we don't officially support Proton, but we do as much as we can in not breaking it and accepting any pull requests that may improve the state.
Yes, I figured. It is not often you hear this as a Linux user, unless it is something official to your distro.

When you label this as "need more info", is this info that I can provide?

**Comment by jonpas** [comment]
> It looks like it coincided just about with the latest Arma update on October 5th.

Nothing changed on our side. Those logs aren't very useful either, they don't indicate a crash at all, so it's probably Proton itself crashing. Look at those logs.

> When you label this as "need more info", is this info that I can provide?

Not necessarily.

**Comment by Tinter** [comment]
Well for good measure, here are the Proton logs.
With ACRE: https://gist.github.com/Tinter/f0a3499b803ae48ec38a1eef5225d8f8
Without ACRE: https://gist.github.com/Tinter/df9ecc0e45efdaa08e62b61b7a4823ea
Output froze at line 3431, lines after that are after I sent sigkill.
About line 3430 for the log that didn't crash. No difference between them that I can see.

**Comment by Freddo3000** [comment]
Experiencing the same issue, but also in multiplayer. Disabling ACRE fixes it.

Running on Linux Mint, latest ACRE, Proton 6.3-7.

**Comment by dedmen** [comment]
Running Arma profiling branch gets you freeze dumps that show what causes the freeze.

**Comment by Tinter** [comment]
> Running Arma profiling branch gets you freeze dumps that show what causes the freeze.

That's a great suggestion, so I have tried it out.
I switched to the profiiling branch under properties->betas.
I have run Arma 3 as I usually would, and also tried to run the profiling binaries, by renaming arma3profiling.exe to arma3.exe (and the same for _x64.exes).
The result is that I can't recreate the issue. My game does not appear to freeze on the profiling branch. 
I've switched to the normal beta to check that it still freezes there, it does.
Switching back again to the profiling branch, it hasn't frozen yet.

So I guess the fix is to run the performance binaries by switching to the profiling branch.
